src/app/policies/views/PolicyListView.vue

```diff
@@ -1,93 +1,170 @@
-        v-if="policyType"
-        class="relative"
-        :class="policyType.path"
+    <DataSource
+      v-slot="{data: policies, error: policyError, refresh: policyRefresh}: PolicyTypeCollectionSource"
+      :src="`/*/policy-types`"
+    >
+      <ErrorBlock
+        v-if="policyError"
+      />
+      <LoadingBlock v-else-if="policies === undefined" />
+      <EmptyBlock v-else-if="policies.policies.length === 0" />
+
+      <template
+        v-else
       >
-        <div class="kcard-stack">
-          <div class="kcard-border">
-            <KCard
-              v-if="policyType.isExperimental"
-              border-variant="noBorder"
-              class="mb-4"
-            >
-              <template #body>
-                <KAlert appearance="warning">
-                  <template #alertMessage>
-                    <p>
-                      <strong>Warning</strong> This policy is experimental. If you encountered any problem please open an
-                      <a
-                        href="https://github.com/kumahq/kuma/issues/new/choose"
-                        target="_blank"
-                        rel="noopener noreferrer"
-                      >issue</a>
-                    </p>
-                  </template>
-                </KAlert>
-              </template>
-            </KCard>
-
-            <DataOverview
-              :selected-entity-name="currentEntityName ?? undefined"
-              :page-size="PAGE_SIZE_DEFAULT"
-              :error="error"
-              :is-loading="isLoading"
-              :empty-state="{
-                title: 'No Data',
-                message: `There are no ${policyType.name} policies present.`,
-              }"
-              :table-data="tableData"
-              :table-data-is-empty="tableData.data.length === 0"
-              :next="nextUrl"
-              :page-offset="pageOffset"
-              @table-action="handleTableAction"
-              @load-data="loadData"
-            >
-              <template #additionalControls>
-                <KSelect
-                  label="Policies"
-                  :items="policySelectItems"
-                  :label-attributes="{ class: 'visually-hidden' }"
-                  appearance="select"
-                  :enable-filtering="true"
-                  @selected="changePolicyType"
-                >
-                  <template #item-template="{ item }">
-                    <span
-                      :class="{
-                        'policy-type-empty': policyTypeNamesWithNoPolicies.includes(item.label)
+        <template
+          v-for="selected in [policies.policies.find(item => item.path === _route.params.policyPath) ?? policies.policies[0]]"
+          :key="selected.path"
+        >
+          <DataSource
+            v-slot="{data, error, refresh}: PolicyCollectionSource"
+            :src="`/${_route.params.mesh}/policy-type/${selected.path}?offset=${route.query.offset || '0'}`"
+            @change="change"
```

Comment:
I don't understand why we react to these change? Is it to navigate to the first policy type as soon as it's loaded?

Reply:
I think this one is good to resolve